CRM/Contribute/BAO/Contribution.php

```diff
@@ -3488,14 +3488,13 @@ public static function isSubscriptionCancelled($contributionId) {
    *
    * @param array $params
    *   Contribution object, line item array and params for trxn.
-   *
-   *
    * @param array $financialTrxnValues
    *
-   * @return null|\CRM_Core_BAO_FinancialTrxn
+   * @throws \CRM_Core_Exception
+   * @throws \CiviCRM_API3_Exception
    */
   public static function recordFinancialAccounts(&$params, $financialTrxnValues = NULL) {
-    $skipRecords = $update = $return = $isRelatedId = FALSE;
```

Comment:
Note the goal is that this function would never be called - different code is used when calling 'AddPayment' and the intent is that this would never be hit

Reply:
@eileenmcnaughton I saw the comment but couldn't find the other code - can you link to it?

Reply:
Basically Payment.create does 'something' - ideally it would be a case of the Contribution.create code internally calling Payment.create & this (awful toxic function) would be eliminated. I've slowly been doing stuff with it.

Note that I also support moving parts of it back to the calling function because it's not really a logical/ well connected function - so the call to the fees function could be done from the calling function.